Gosu/Image.h

```diff
@@ -33,12 +33,16 @@ void Gosu_Image_draw(Gosu_Image *image, double x, double y, double z, unsigned c
 void Gosu_Image_draw_rot(Gosu_Image *image, double x, double y, double z,
                           double angle, double center_x, double center_y,
                           double scale_x, double scale_y, unsigned color, int mode);
-void Gosu_Image_draw_as_quad();
+void Gosu_Image_draw_as_quad(Gosu_Image* image, double x1, double y1, unsigned color1,
+                               double x2, double y2, unsigned color2,
+                               double x3, double y3, unsigned color3,
+                               double x4, double y4, unsigned color4,
+                               double z, int mode);
 
 // Operations
 void Gosu_Image_insert(Gosu_Image *image, Gosu_Image *source, int x, int y);
 void Gosu_Image_save(Gosu_Image *image, char *filename);
```

Comment:
I guess `Image_save` would be a good first example of a function that will probably benefit from having some kind of error code, even if it is only success and failure.

Reply:
I wonder if specific error codes would even help, or if it would be easier to just wrap everything that can fail in `try...catch`, store the exception message in a `static thread_local std::string` somewhere, and just return `true/false` from failable methods.